libvcx/src/aries/handlers/connection/connection.rs

```diff
@@ -684,13 +709,31 @@ Get messages received from connection counterparty.
             agent_vk: cloud_agent_info.agent_vk,
         };
         let object = SerializableObjectWithState::V1 { data, state, source_id };
-        json!(object).to_string()
+        serializer.serialize_some(&object)
     }
+}
 
-    pub fn from_string(connection_data: &str) -> VcxResult<Connection> {
-        let object: SerializableObjectWithState<LegacyAgentInfo, SmConnectionState> = serde_json::from_str(connection_data)
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize Connection: {:?}", err)))?;
-        match object {
+struct ConnectionVisitor;
+
+impl<'de> Visitor<'de> for ConnectionVisitor {
+    type Value = Connection;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an integer between -2^31 and 2^31")
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, <A as MapAccess<'de>>::Error> where
+        A: MapAccess<'de>, {
+        let mut map_value = serde_json::Map::new();
+        while let Some(key) = map.next_key()? {
+            let k: String = key;
+            let v: Value = map.next_value()?;
+            map_value.insert(k, v);
+        }
+        let obj = Value::from(map_value);
+        let ver: SerializableObjectWithState<LegacyAgentInfo, SmConnectionState> = serde_json::from_value(obj)
```

Comment:
Removing `LegacyAgentInfo` is still very soon on the roadmap, correct?

Reply:
yes yes :-D Will do that as separate breaking change PR